pkg/helm/release/manager.go

```diff
@@ -194,42 +145,25 @@ func (m manager) getDeployedRelease() (*rpb.Release, error) {
 	return deployedRelease, nil
 }
 
-func (m manager) getCandidateRelease(ctx context.Context, tiller *tiller.ReleaseServer, name string, chart *cpb.Chart, config *cpb.Config) (*rpb.Release, error) {
-	dryRunReq := &services.UpdateReleaseRequest{
-		Name:   name,
-		Chart:  chart,
-		Values: config,
-		DryRun: true,
-	}
-	dryRunResponse, err := tiller.UpdateRelease(ctx, dryRunReq)
-	if err != nil {
-		return nil, err
-	}
-	return dryRunResponse.GetRelease(), nil
+func (m manager) getCandidateRelease(ctx context.Context, namespace, name string, chart *cpb.Chart, values map[string]interface{}) (*rpb.Release, error) {
+	upgrade := action.NewUpgrade(m.actionConfig)
+	upgrade.Namespace = namespace
+	upgrade.DryRun = true
+	return upgrade.Run(name, chart, values)
 }
 
 // InstallRelease performs a Helm release install.
 func (m manager) InstallRelease(ctx context.Context) (*rpb.Release, error) {
-	return installRelease(ctx, m.tiller, m.namespace, m.releaseName, m.chart, m.config)
-}
+	install := action.NewInstall(m.actionConfig)
+	install.ReleaseName = m.releaseName
+	install.Namespace = m.namespace
 
-func installRelease(ctx context.Context, tiller *tiller.ReleaseServer, namespace, name string, chart *cpb.Chart, config *cpb.Config) (*rpb.Release, error) {
-	installReq := &services.InstallReleaseRequest{
-		Namespace: namespace,
-		Name:      name,
-		Chart:     chart,
-		Values:    config,
-	}
-
-	releaseResponse, err := tiller.InstallRelease(ctx, installReq)
+	installedRelease, err := install.Run(m.chart, m.values)
 	if err != nil {
 		// Workaround for helm/helm#3338
```

Comment:
Is this comment and related code still relevant? The issue is closed.

Ditto below.

Reply:
I tried using the new `Atomic` config item, but that changed the semantics in a way I didn't expect. IIRC, it also forces `Wait`, which cascaded into some other issues. I think we should revisit this, but maybe in a follow-up?